### PR TITLE
Viewport UI focus/shortut fix

### PR DESCRIPTION
**Pull request by hultonha**
This is a fix for focus being removed from the main viewport widget when interacting with viewport ui elements. This would result in viewport shortcuts not firing (e.g. R to reset rotation/scale etc...) after having clicked a viewport ui element, it would be necessary to click back in the viewport again to fix things (an incredibly irritating bug).

The crux of the fix is in `ViewportUiDisplay.cpp` - we now pass `Qt::WindowDoesNotAcceptFocus` to the `windowFlags` in `ConfigureWindowForViewportUi` (see below)

```
static void ConfigureWindowForViewportUi(QPointer<QMainWindow> mainWindow)
{
    // no background for the widget else each set of buttons/text-fields/etc would have a black box around them
    SetTransparentBackground(mainWindow);
    mainWindow->setWindowFlags(Qt::Window | Qt::FramelessWindowHint | Qt::WindowDoesNotAcceptFocus);
}
```

I'm going to also look into writing an integration test (we already have a few for Viewport Ui) to see if we can reproduce and verify the fix. This investigation took a while though so I'd like to get this posted as-is for now. There's also a few other minor fixes and some formatting updates (using `clang-format`).

### CODE REVIEW

File: Code/Framework/AzToolsFramework/AzToolsFramework/ViewportUi/ViewportUiDisplay.cpp [hunk 1]
```diff
@@ -329,48 +329,25 @@ namespace AzToolsFramework::ViewportUi::Internal
...
     // disables system background for widget and gives a transparent background
-    static void ConfigureWidgetForViewportUi(QPointer<QWidget> widget)
+    static void ConfigureWindowForViewportUi(QPointer<QMainWindow> mainWindow)
     {
-        // no background for the widget else each set of buttons/textfields/etc would have a black box around them
-        SetTransparentBackground(widget);
-        widget->setWindowFlags(Qt::Window | Qt::FramelessWindowHint);
+        // no background for the widget else each set of buttons/text-fields/etc would have a black box around them
+        SetTransparentBackground(mainWindow);
+        mainWindow->setWindowFlags(Qt::Window | Qt::FramelessWindowHint | Qt::WindowDoesNotAcceptFocus);
```
Comment (NicholasVanSickle) [inline Code/Framework/AzToolsFramework/AzToolsFramework/ViewportUi/ViewportUiDisplay.cpp:344]: So I don't lose it again: the actual fix is here.

**Review by NicholasVanSickle — APPROVED** [review]
Looks good to me, great job tracking this down!

Suggestion for future PRs: if you're going to do a formatting pass in a PR, it'd be very helpful to split your branch into atomic commits corresponding to the refactoring and the actual fix. This can make a `git log` or `git annotate` a lot more helpful if you do a Merge instead of a Squash, as well as making the review easier by allowing reviewers to go through the individual commits. Cheers!

File: Code/Framework/AzToolsFramework/AzToolsFramework/ViewportUi/ViewportUiDisplay.cpp (hunk 1 above)
Comment (hultonha) [reply Code/Framework/AzToolsFramework/AzToolsFramework/ViewportUi/ViewportUiDisplay.cpp:344]: Totally! Yes sorry I do usually try to do this I think I just think I wound up poking around so much and wanted to fix all these little bits and bobs I saw..

Maybe I can re-post this as two separate PRs so I follow my own rules! 😝 (I think I was just so elated I fixed it I just wanted to post the fix ASAP 😅)

Comment (AMZN-daimini) [reply Code/Framework/AzToolsFramework/AzToolsFramework/ViewportUi/ViewportUiDisplay.cpp:344]: You are forgiven.

**Review by AMZN-daimini — APPROVED** [review]

**Comment by hultonha** [comment]
I'm going to close this and post a new split PR - one with the core fix, and one with the other clean-up fixes (thanks for the approvals though! 🙂

**Pull request closed without merging**